index.js

```diff
@@ -46,21 +30,29 @@ module.exports._reset = function () {
  */
 module.exports.releaseExit = function() {
   if (exit) {
-    process.exit = exit;
+    _process.exit = exit;
     exit = null;
   }
 };
 
 var firstExitCode;
-
-module.exports.captureExit = function() {
+module.exports.captureExit = function(process) {
   if (exit) {
     // already captured, no need to do more work
     return;
   }
-  exit = process.exit;
 
-  process.exit = function(code) {
+  if (process instanceof EventEmitter === false) {
+    throw new TypeError('attempt to capture a bad process instance');
+  }
+
+  _process = process;
```

Comment:
There are at least two places where we need reference to `_process`:
 - at least for `onBeforeExit` [cleanup](https://github.com/ro0gr/capture-exit/blob/9ed93186b7a66e9a133d48cc68bcfad9d05c6631/index.js#L138) (added in last commit)
 - [`exit` proxy](https://github.com/ro0gr/capture-exit/blob/9ed93186b7a66e9a133d48cc68bcfad9d05c6631/index.js#L142)